DesktopEdge/Models/ZitiService.cs

```diff
@@ -121,13 +121,33 @@ public class ZitiService {
 
 		public Visibility TimerVisibility {
 			get {
-				return (TimeoutRemaining>-1&&TimeoutRemaining<1260) ? Visibility.Visible : Visibility.Collapsed;
+				return (TimeoutCalculated > -1&& TimeoutCalculated < 1260&& TimeoutCalculated > 0) ? Visibility.Visible : Visibility.Collapsed;
+			}
+			set { }
+		}
+
+		public int TimeoutCalculated { 
+			get {
+				if (this.TimeoutRemaining == -1|| TimeoutRemaining == 0) return this.TimeoutRemaining;
+				else {
+					TimeSpan t = (DateTime.Now - this.TimeUpdated);
+					int timeout = this.Timeout - (int)Math.Floor(t.TotalSeconds);
+					if (timeout < 0) timeout = 0;
+					return timeout;
+				}
+			}
+			set { }
+		}
+
+		public Visibility MfaVisibility {
+			get {
+				return (TimeoutCalculated > -1 && TimeoutCalculated == 0) ? Visibility.Visible : Visibility.Collapsed;

```

Comment:
personally I think `TimeoutCalculated > -1 && TimeoutCalculated == 0` should be a function that makes it easy to call, set and read... Something like `IsMfaTimeoutNeeded` or someshuch?

Reply:
These are UI Properties that are only there to show or hide visuals. They do not match, there is three different states, MFA Lock, MFA Timer, and none.